### PR TITLE
[FIX] mail: fix activity view domain on reload

### PR DESCRIPTION
**Pull request by kcv-odoo**
Steps:
- Install project app.
- Go to task view from specific project.
- Open activity view.
- Set an activity on any task.
- Click on "Schedule Activity" button.

Issue:
- User able to select all tasks instead of tasks from that specific project as it is working for first time user opens that view.

Cause:
- Domain of `SelectCreateDialog` became empty when load calls from other activity view actions.

Fix:
- Pass props in load method call to have default data as it is when calling load method.

task-3293310

### CODE REVIEW

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

I will let @odoo/rd-framework-js doing the final review. 🙂

**Comment by FrancoisGe** [comment]
> Steps:
> 
> * Install project app.
> * Go to task view from specific project.
> * Open activity view.
> * Set an activity on any task.
> * Click on "Schedule Activity" button.
> 
> Issue:
> 
> * User able to select all tasks instead of tasks from that specific project as it is working for first time user opens that view.
> 
> Cause:
> 
> * Domain of `SelectCreateDialog` became empty when load calls from other activity view actions.
> 
> Fix:
> 
> * Pass Original domain in load action call to have domain as it is.
> 
> task-3293310
> 
> Description of the issue/feature this PR addresses:
> 
> Current behavior before PR:
> 
> Desired behavior after PR is merged:
> 
> I confirm I have signed the CLA and read the PR guidelines at [www.odoo.com/submit-pr](http://www.odoo.com/submit-pr)

Hello @kcv-odoo 
I'm not sure that your fix solves all the problems, as it doesn't take account of contexts, etc. 
Here's an idea for a more in-depth fix: https://github.com/odoo-dev/odoo/commit/c2adb7040aa63dfe853935069d89f9008fe591c9

**Comment by FrancoisGe** [comment]
You shouldn't use jquery and legacy utils in your test :)

**Comment by kcv-odoo** [comment]
> Hello @kcv-odoo I'm not sure that your fix solves all the problems, as it doesn't take account of contexts, etc. Here's an idea for a more in-depth fix: [odoo-dev@c2adb70](https://github.com/odoo-dev/odoo/commit/c2adb7040aa63dfe853935069d89f9008fe591c9)

Hello @FrancoisGe 
Ohh I see, I adapted the changes to take other thing into account
Also remove jquery and legacy utils use from code, should be fine now :)

Thank you for the suggestion :innocent:

File: addons/test_mail/static/tests/activity_tests.js [hunk 1]
```diff
@@ -623,6 +623,47 @@ QUnit.module("test_mail", {}, function () {
...
+            await openView({
+                res_model: "mail.test.activity",
+                views: [[false, "activity"]],
+                domain: [['id', '=', 1]],
+            });
+            patchWithCleanup(env.services.action, {
+                doAction(action, options) {
+                    assert.step("doAction");
+                    options.onClose();
+                    return Promise.resolve();
```
Comment (FrancoisGe) [inline addons/test_mail/static/tests/activity_tests.js:648]: You don t need to return something. I think

File: addons/test_mail/static/tests/activity_tests.js [hunk 2]
```diff
@@ -623,6 +623,47 @@ QUnit.module("test_mail", {}, function () {
...
 
+    QUnit.test("activity view: Domain should not reset on load", async function (assert) {
+            Object.assign(serverData.views, {
+                "mail.test.activity,false,list":
+                    '<tree string="MailTestActivity"><field name="name"/></tree>',
+            });
+            const { env, openView } = await start({
+                mockRPC(route, args) {
+                    if (args.method === "name_search") {
+                        args.kwargs.name = "MailTestActivity";
```
Comment (FrancoisGe) [inline addons/test_mail/static/tests/activity_tests.js:634]: Do you need to edit args.kwargs.name ? It s not possible do to do this research with the interface and receive the good value in args.kwargs.name?

Comment (kcv-odoo) [reply addons/test_mail/static/tests/activity_tests.js]: No I guess, Bad copy paste :cry:

**Comment by FrancoisGe** [comment]
@robodoo r+

**Comment by FrancoisGe** [comment]
Good job :)

**Pull request closed without merging**